cpp/memilio/math/floating_point.h

```diff
@@ -72,7 +72,7 @@ template <class T>
 bool floating_point_less(T v1, T v2, T abs_tol = 0, T rel_tol = std::numeric_limits<T>::min())
 {
     auto diff = v1 - v2;
-    return diff < -abs_tol || diff < -abs_max(v1, v2) * rel_tol;
+    return diff < abs_tol || diff < abs_max(v1, v2) * rel_tol;
```

Comment:
I think the change is not correct as now `a) v1 not == v2 within tolerances` is not satisfied anymore.

Reply:
You are right, I misunderstood the function. But this means that increasing a tolerance *reduces* how "tolerant" the function is, correct? I think this is misleading.

Moreover, I think the old version does not work correctly as well: Take `v1=1.1, v2=3.2, abs_tol=20 , rel_tol=0`. Then we have
```
floating_point_less(v1, v2, abs_tol, 0.0) == true;
```
since ` -2.1 < -0` from the rel_tol comparision holds true, but the values are certainly equal w.r.t abs_tol. Is this as intended?

Reply:
I think the correct version might be:
```cpp
bool floating_point_less(T v1, T v2, T abs_tol = 0, T rel_tol = std::numeric_limits<T>::min())
{
    auto diff = v1 - v2;
    return diff < -(abs_tol + rel_tol * abs_max(v1, v2));
}
```
I already pushed it, and in the tests there is a truth table that shows how this behaves. Could you please double check that this has the intended behavior?

Reply:
At least with this version, `floating_point_less_equal == !floating_point_greater` and `floating_point_greater_equal == !floating_point_less` hold.

Reply:
perfect